tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js

```diff
@@ -0,0 +1,335 @@
+        this,
+        'testIdentifier',
+        'goToDiscountsPage',
+        baseContext,
+      );
+
+      await dashboardPage.goToSubMenu(
+        page,
+        dashboardPage.catalogParentLink,
+        dashboardPage.discountsLink,
+      );
+
+      const pageTitle = await cartRulesPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(cartRulesPage.pageTitle);
+    });
+
+    it('should go to new cart rule page', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'goToNewCartRulePage',
+        baseContext,
+      );
+
+      await cartRulesPage.goToAddNewCartRulesPage(page);
+
+      const pageTitle = await addCartRulePage.getPageTitle(page);
+      await expect(pageTitle).to.contains(addCartRulePage.pageTitle);
+    });
+
+    it('should create cart rule', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'createCartRule',
+        baseContext,
+      );
+
+      const validationMessage = await addCartRulePage.createEditCartRules(page, cartRuleCode);
+      await expect(validationMessage).to.contains(addCartRulePage.successfulCreationMessage);
+    });
+
+    it('should view my shop', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'viewMyShop1',
+        baseContext,
+      );
+
+      // View my shop and init pages
+      page = await addCartRulePage.viewMyShop(page);
+
+      await foHomePage.changeLanguage(page, 'en');
+      const isHomePage = await foHomePage.isHomePage(page);
+      await expect(isHomePage, 'Fail to open FO home page').to.be.true;
+    });
+  });
+
+  const tests = [
+    {args: {testIdentifier: 'cartRuleAccepted', testTitle: 'for the first time'}},
+  ];
+
+  tests.forEach((test) => {
+    describe(`Use Cart Rule ${test.args.testTitle}`, async () => {
+      it('should go to the first product page', async function () {
+        await testContext.addContextItem(
+          this,
+          'testIdentifier', `
+          ${test.args.testIdentifier}GoToFirstProductPage`,
+          baseContext,
+        );
+
+        await foHomePage.goToProductPage(page, 1);
+
+        const pageTitle = await foProductPage.getPageTitle(page);
+        await expect(pageTitle.toUpperCase()).to.contains(ProductData.firstProductData.name);
+      });
+
+      it('should add product to cart and proceed to checkout', async function () {
+        await testContext.addContextItem(
+          this,
+          'testIdentifier',
+          `${test.args.testIdentifier}AddProductToCart`,
+          baseContext,
+        );
+
+        await foProductPage.addProductToTheCart(page);
+
+        const notificationsNumber = await cartPage.getCartNotificationsNumber(page);
+        await expect(notificationsNumber).to.be.equal(1);
+      });
+
+      it('should get "You must choose a carrier before applying this voucher" error message',
```

Comment:
In this iteration you add a promo code and you don't get "You must choose a carrier before applying this voucher" error message

Reply:
DONE